src/gltf/GltfModelObjectMetaData.hpp

```diff
@@ -0,0 +1,139 @@
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+#ifndef GLTF_GLTFMODELOBJECTMETADATA_HPP
+#define GLTF_GLTFMODELOBJECTMETADATA_HPP
+
+#include "GltfAPI.hpp"
+
+#include "../utilities/core/Logger.hpp"
+
+namespace tinygltf {
+class Value;
+}
+
+namespace openstudio {
+namespace model {
+
+  class AirLoopHVAC;
+  class BuildingUnit;
+  class SpaceType;
+  class BuildingStory;
+  class ThermalZone;
+  class DefaultConstructionSet;
+  class Space;
+}  // namespace model
+
+namespace gltf {
+
+  class GltfMetaData;
+
+  /** GltfModelObjectMetaData is an Interface class between select ModelObjects and glTF values
+   *
+   *  It is capable of storing attributes such as name, iddObjectType, color, handle from ModelObjects:
+   *  ThermalZone, SpaceType, AirLoopHVAC, BuildingStory, BuildingUnit, DefaultConstructionSet
+   **/
+  class GLTF_API GltfModelObjectMetaData
+  {
+   public:
+    /** Default constructor **/
+    GltfModelObjectMetaData();
+
+    /** @name Interface with ModelObjects */
+    //@{
+    explicit GltfModelObjectMetaData(const model::AirLoopHVAC& airLoopHVAC);
+    explicit GltfModelObjectMetaData(const model::BuildingUnit& buildingUnit);
+    explicit GltfModelObjectMetaData(const model::SpaceType& spaceType);
+    explicit GltfModelObjectMetaData(const model::BuildingStory& buildingStory);
+    explicit GltfModelObjectMetaData(const model::ThermalZone& thermalZone);
+    explicit GltfModelObjectMetaData(const model::DefaultConstructionSet& defaultConstructionSet);
+    explicit GltfModelObjectMetaData(const model::Space& space);
+    //@}
+
+    /** @name Getters and Setters */
+    //@{
+
+    std::string color() const;
+    void setColor(const std::string& color);
+
+    std::string handle() const;
+    void setHandle(const std::string& handle);
+
+    std::string iddObjectType() const;
+    void setIddObjectType(const std::string& iddObjectType);
+
+    std::string name() const;
+    void setName(const std::string& name);
+
+    bool isOpentoBelow() const;
+    void setOpentoBelow(bool opentoBelow);
+
+    double nominalZCoordinate() const;
+    void setNominalZCoordinate(double nominalZCoordinate);
+
+    double nominalFloortoCeilingHeight() const;
+    void setNominalFloortoCeilingHeight(double nominalFloortoCeilingHeight);
+
+    int multiplier() const;
+    void setMultiplier(int multiplier);
+
+    //@}
+
+   protected:
+    /** @name Protected */
+    //@{
+
+    /** Interface with an existing GLTF file **/
+    explicit GltfModelObjectMetaData(const tinygltf::Value& modelObjectMetaDataObj);
+
+    /** Export to tinygltf **/
+    std::map<std::string, tinygltf::Value> toExtras() const;
```

Comment:
Protected area: 

The Ctor  is to do the reverse: populate from a GLTF object

The toExtras exports to a tinygltf object